BeerOverflowWindowsApp/BarRating.cs

```diff
@@ -8,92 +8,90 @@ namespace BeerOverflowWindowsApp
             }
-            BarFileWriter.SaveData(_barsData);
+            BarFileWriter.SaveData(allBars);
         }
 
         public void AddBars(List<BarData> barsList)
         {
             foreach (var bar in barsList)
             {
-                if (_barsData.BarsList.Count(x => x.Title == bar.Title) == 0)
+                if (BarsData.BarsList.Count(x => x.Title == bar.Title) == 0)
                 {
-                    _barsData.BarsList.Add( bar );
+                    BarsData.BarsList.Add( bar );
                 }
             }
-            BarFileWriter.SaveData(_barsData);
+            BarFileWriter.SaveData(BarsData);
         }
 
         public void Sort(CompareType compareType)
         {
             switch (compareType)
             {
                 case CompareType.Title:
-                    var titleComparer = new ComparerByTitle();
-                    _barsData.BarsList.Sort(titleComparer);
-                    if (SortByTitleDesc)
-                    {
-                        _barsData.BarsList.Reverse();
-                    }
-                    SortByTitleDesc = !SortByTitleDesc;
-                    SortByRatingDesc = false;
-                    SortByDistance = false;
+                    SortAndInvertIfNeeded(new ComparerByTitle(), CompareType.Title);
                     break;
                 case CompareType.Rating:
-                    var ratingsComparer = new ComparerByRating();
-                    _barsData.BarsList.Sort(ratingsComparer);
-                    if (SortByRatingDesc)
-                    {
-                        _barsData.BarsList.Reverse();
-                    }
-                    SortByTitleDesc = false;
-                    SortByRatingDesc = !SortByRatingDesc;
-                    SortByDistance = false;
+                    SortAndInvertIfNeeded(new ComparerByRating(), CompareType.Rating);
                     break;
                 case CompareType.Distance:
-                    var distanceComparer = new ComparerByDistance();
-                    _barsData.BarsList.Sort(distanceComparer);
-                    if (SortByDistance)
-                    {
-                        _barsData.BarsList.Reverse();
-                    }
-                    SortByTitleDesc = false;
-                    SortByRatingDesc = false;
-                    SortByDistance = !SortByDistance;
+                    SortAndInvertIfNeeded(new ComparerByDistance(), CompareType.Distance);
                     break;
                 default:
                     throw new ArgumentOutOfRangeException(nameof(compareType), compareType, null);
             }
         }
+
+        private void SortAndInvertIfNeeded(IComparer<BarData> comparer, CompareType compareType)
+        { 
+            if (BarsData != null)
+            {
+                BarsData.BarsList.Sort(comparer);
+                if (compareType == _lastCompare)
```

Comment:
This logic works but is not the best one. Try multiple searches of the same coordinates, the list inverts everytime.

Reply:
Fixed inversion